cl/api/management/commands/cl_retry_webhooks.py

```diff
@@ -17,16 +18,52 @@ def retry_webhook_events() -> int:
 
     # Using select_for_update() and transaction.atomic() here will prevent more
     # than one instance retry pending webhooks, avoiding duplicated retries.
-    webhook_events = WebhookEvent.objects.select_for_update().filter(
-        event_status=WEBHOOK_EVENT_STATUS.ENQUEUED_RETRY,
+    cut_off_date_two_days = now() - timedelta(days=2)
```

Comment:
I like the approach you're moving towards here, but it still has two (small) problems:

 - It separates the logic of the query from the action of it. In other words, you define the query up on line 21, but use it down on line 58. You have to scroll up and down to see what the code is doing.
 - It unnecessarily obfuscates (hides) the differences between the three queries, when they're actually quite similar. 

What about something like this:

```
with transaction.atomic():
    base_events = WebhookEvent.objects.filter(
        next_retry_date__lte=now(),
        webhook__enabled=True,
        debug=False,
    ).order_by("date_created")
    failed_webhook_events = base_events.filter(
        event_status__in=[
            WEBHOOK_EVENT_STATUS.ENQUEUED_RETRY,
            WEBHOOK_EVENT_STATUS.ENDPOINT_DISABLED,
        ],
        date_created__lt=cut_off_date_two_days,
    ).update(
        event_status=WEBHOOK_EVENT_STATUS.FAILED, 
        date_modified=now()
    )
    webhook_events_to_restart = base_events.filter(
        ...
    ).update(
        ...
    )
    webhook_events_to_retry = base_events.filter(...)
    for webhook_event in webhook_events_to_retry:
            send_webhook_event(webhook_event)
```

I think this removes the need for the `select_for_update` too?

Reply:
Thanks, this approach makes things simpler and clearer, changed. 
Yes, this removes the need for `select_for_update` in all the child queries, I only keep it in the base query. It's supposed we'll have only one machine running this command but in case for any reason, we have two or more instances running it, this would prevent for possible race conditions.